Add support for TikTok Archive JSONs

Jadoo4QFan opened the pull request:
This code was tested upon two personal exports I had, one from late 2023 and one from early 2025. Additionally it was tested on the freely available exports from https://www.google.com/search?q=%22FavoriteHashtagList%22 .

Jadoo4QFan committed:
Create tikTokArchiveJson.py

Jadoo4QFan committed:
Update tikTokArchiveJson.py

Jadoo4QFan committed:
Update tikTokArchiveJson.py

abrignoni commented:
Per repository readme the project is using the artifacts V2 header and not
a footer anymore. Please update. Thanks for the contribution.

On Mon, Jun 16, 2025, 1:48 AM Jadoo4QFan ***@***.***> wrote:

> *Jadoo4QFan* left a comment (abrignoni/RLEAPP#250)
> <https://github.com/abrignoni/RLEAPP/pull/250#issuecomment-2975184486>
>
> Sources (other than personal exports):
>
> https://gist.github.com/extratone/c6c81f022721ee9f5a85bbdc28136661
>
> https://github.com/zollo/tiktok-downloader/blob/main/data_structure.yml
>
>
> https://gitlab-10.mpi-sws.org/ns/computational-social-sience/tiktok-playlist-experiment/-/blob/6c0742493025d70a4dcb589262df53e5955fe746/user_gdpr_data/mpisws43-2109.json
>
> https://www.scribd.com/document/863088396/User-Data-Tiktok (never used)
>
> ToDo: support ~June 2025 exports
>
>
> https://github.com/mervat-e/mervat-e.github.io/blob/1c00dfdbcdc9cb98c09287ea30148744e78d0cc3/frontend/modified_tiktok.json#L23
>
>
> https://github.com/mervat-e/mervat-e.github.io/blob/1c00dfdbcdc9cb98c09287ea30148744e78d0cc3/backend/resources/user_data/tiktok/merged_tiktok_user.json#L23
>
> Potential artifacts: Activity Summary, Most Recent Location Data
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/abrignoni/RLEAPP/pull/250#issuecomment-2975184486>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG3DPC2QNAX3FJ75XZNIT2T3DZLC3AVCNFSM6AAAAAB7MABULKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNZVGE4DINBYGY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

Jadoo4QFan committed:
Update tikTokArchiveJson.py

Jadoo4QFan committed:
Support mid-2025 archives, add placeholder artifacts

Tested multiple times with trial-and-error with a late-2023, January 2025, and Summer 2025 export.

Jadoo4QFan commented:
I'd say it's ready to be merged now.
